lib/Runtime/Library/JavascriptArray.inl

```diff
@@ -129,6 +131,12 @@ namespace Js
         {
             segmentMap->SwapSegment(seg->left, seg, newSeg);
         }
+
+        if (isInlineSegment)
+        {
+            this->ClearElements(seg, 0);
+        }
```

Comment:
I think the inline segment is a plus alloc, it's part of the array structure, you can't free only a part of memory of one allocation